### ISSUE TITLE
[BUG] Kotlin-spring generator generates invalid project

### ISSUE THREAD

**Issue by thawkins** [OpenAPITools/openapi-generator#4368]
##### Description

Generating kotlin-spring project produces service that does not compile

I'm trying to generate a SCIM2 srv with attached Openapi definition

generating with
$mkdir outputtest 
$openapi-generator generate -i  ./scim2svc.yml -g kotlin-spring -o outputtest
$cd outputtest
$gradle wrapper
$./gradlew clean
$./gradlebuild build 

this fails with the errors

```
[tim.hawkins@localhost scim2svc]$ ./gradlew build

> Task :compileKotlin FAILED
e: /home/tim.hawkins/IdeaProjects/blackswan/services/airsoc/scim2svc/src/main/kotlin/org/openapitools/model/ErrorResponse.kt: (38, 147): Type mismatch: inferred type is String but List<String> was expected
e: /home/tim.hawkins/IdeaProjects/blackswan/services/airsoc/scim2svc/src/main/kotlin/org/openapitools/model/SearchRequest.kt: (38, 163): Type mismatch: inferred type is String but List<String> was expected
e: /home/tim.hawkins/IdeaProjects/blackswan/services/airsoc/scim2svc/src/main/kotlin/org/openapitools/model/ServiceProviderConfig.kt: (57, 179): Type mismatch: inferred type is String but List<String> was expected
e: /home/tim.hawkins/IdeaProjects/blackswan/services/airsoc/scim2svc/src/main/kotlin/org/openapitools/model/User.kt: (69, 145): Type mismatch: inferred type is String but List<String> was expected
e: /home/tim.hawkins/IdeaProjects/blackswan/services/airsoc/scim2svc/src/main/kotlin/org/openapitools/model/UserPatch.kt: (35, 151): Type mismatch: inferred type is String but List<String> was expected
e: /home/tim.hawkins/IdeaProjects/blackswan/services/airsoc/scim2svc/src/main/kotlin/org/openapitools/model/UserResponse.kt: (75, 145): Type mismatch: inferred type is String but List<String> was expected
e: /home/tim.hawkins/IdeaProjects/blackswan/services/airsoc/scim2svc/src/main/kotlin/org/openapitools/model/UserSearchResponse.kt: (42, 161): Type mismatch: inferred type is String but List<String> was expected

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileKotlin'.
> Compilation error. See log for more details

```
##### openapi-generator version

```
[tim.hawkins@localhost scim2svc]$ openapi-generator version
4.1.3
[tim.hawkins@localhost scim2svc]$ 
```

##### OpenAPI declaration file content or url

[scim2svc.zip](https://github.com/OpenAPITools/openapi-generator/files/3807286/scim2svc.zip)

##### Command line used for generation

generating with
$mkdir outputtest 
$openapi-generator generate -i  ./scim2svc.yml -g kotlin-spring -o outputtest
$cd outputtest
$gradle wrapper
$./gradlew clean
$./gradlebuild build 

##### Steps to reproduce

use generator to generate project into a clean output directory

##### Related issues/PRs

None found

##### Suggest a fix

Not known

**Comment by thawkins** [comment]
Further information

All the errors are related to the use of this construct 

```
     schemas:
        type: "array"
        items:
          type: "string"
          enum:
          - "urn:ietf:params:scim:api:messages:2.0:Error"
```

changing this to:

```
schemas:
    type: "string"
       enum:
       - "urn:ietf:params:scim:api:messages:2.0:Error"
```

fixes the problem, however it means that the data class generated only has a single string allocated for Schemas, where as the intent was to have a List of schemas. I thinbk the generator chokes on an array of strings where there is only and enum value with a single entry. 

With the changed input it creates:
```
/**
    * 
    * Values: urnColonIetfColonParamsColonScimColonApiColonMessagesColon2Period0ColonSearchRequest
    */
    enum class Schemas(val value: kotlin.String) {
    
        @JsonProperty("urn:ietf:params:scim:api:messages:2.0:SearchRequest") urnColonIetfColonParamsColonScimColonApiColonMessagesColon2Period0ColonSearchRequest("urn:ietf:params:scim:api:messages:2.0:SearchRequest");
    
    } 
```

with the original structure it creates 

```
/**
    * 
    * Values: urnColonIetfColonParamsColonScimColonApiColonMessagesColon2Period0ColonSearchRequest
    */
    enum class Schemas(val value: List<kotlin.String>) {
    
        @JsonProperty("urn:ietf:params:scim:api:messages:2.0:SearchRequest") urnColonIetfColonParamsColonScimColonApiColonMessagesColon2Period0ColonSearchRequest("urn:ietf:params:scim:api:messages:2.0:SearchRequest");
    
    } 
```

**Comment by nekkiy** [comment]
fixed in https://github.com/OpenAPITools/openapi-generator/pull/4565

**Comment by jimschubert** [comment]
closed via #4977, #4565 

Thank you for the bug report and the analysis about the root cause. I've verified that #4565 does indeed resolve issues around arrays of embedded and referenced enums.

With that change, you will now see a list of Schemas properly referencing the embedded class:


```
data class UserSearchResponse (
        @JsonProperty("schemas") val schemas: kotlin.collections.List<UserSearchResponse.Schemas>? = null
) {
    enum class Schemas(val value: kotlin.String) {
        @JsonProperty("urn:ietf:params:scim:api:messages:2.0:ListResponse") urnColonIetfColonParamsColonScimColonApiColonMessagesColon2Period0ColonListResponse("urn:ietf:params:scim:api:messages:2.0:ListResponse");
    
    }

}
```

I'm not sure if there's support for modifying the enum name currently, but this looks structurally as I'd expect. Since I've merged the fix, I'll go ahead and close this. Please feel free to reopen if there are additional concerns after evaluating the latest master.
